xray/conventions.py

```diff
@@ -807,6 +807,10 @@ def stackable(dim):
 
     new_vars = OrderedDict()
     for k, v in iteritems(variables):
+        if isinstance(drop_variables, basestring):
```

Comment:
Let's also convert `drop_variables` to a set. That will keep things performant, even if the dataset has a very large number of variables.